### DISCUSSION TITLE
NoMethodError: undefined method `rewind'

### DISCUSSION THREAD

**Discussion by Isuresh** [ThrowTheSwitch/Ceedling#816] [Q&A]
Hello , 

I ran into below  problem while running test cases , earlier it worked fine can some one help to resolve issue . 
![Ceedling_error](https://github.com/ThrowTheSwitch/Ceedling/assets/36395674/63d0da47-b673-4b8a-8736-cc0602ea9512)

Thanks in Advance

**Comment by mvandervoord** [comment]
Is it too late to capture the rest of that error message? It'd be good to get the rest of the stacktrace to see where that error happened.

**Reply by Isuresh** [reply]
Hi @mvandervoord , 

There is No errors before that .. please find complete result . 

** Execute build/test/runners/test_batt_state_runner.c
rake aborted!
****NoMethodError: undefined method `rewind' for "#include** \**"mock_app_alarm_core.h\"\n#include \"mock_hal_gpio.h\"\n#include \"mock_drv_accel.h\"\n#include \"mock_lib_noninit.h\"\n#include \"mock_thread_timer.h\"\n#include \"mock_thread_test.h\"\n#include \"mock_crc.h\"\n#include \"mock_drv_usbd.h\"\n#include \"mock_debug.h\"\n#include \"mock_hal_adc.h\"\n#include \"mock_thread_alarm.h\"\n#include \"mock_app_led.h\"\n#include \"mock_thread_led.h\"\n#include \"mock_drv_rtc.h\"\n#include \"mock_app_log.h\"\n#include \"mock_thread_mode.h\"\n#include \"mock_thread_types.h\"\n#include \"mock_thread_wdt.h\"\n#include \"mock_lib_csr.h\"\n#include \"mock_app_csr.h\"\n#include \"mock_RTOS.h\"\n#include \"mock_utils.h\"\n#include \"mock_p_assert.h\"\n#include \"mock_rtal.h\"\n#include \"mock_thread_batt.h\"\n#include \"mock_nordic_common.h\"\n#include \"mock_sdk_config.h\"\n#include \"../Application/batt_state.h\"\n#include \"../Platform/nRF5_SDK_Modded/s113/headers/nrf_soc.h\"\n#include \"../Platform/nRF5_SDK_Modded/s113/headers/nrf_svc.h\"\n#include \"../../Common/nRF5_SDK/modules/nrfx/hal/nrf_power.h\"\n#include \"../Application/Algorithm/Algorithm_1.9/Algorithm_FW/local_types.h\"\n#include \"../../Common/nRF5_SDK/components/toolchain/cmsis/include/cmsis_gcc.h\"\n#include \"../../Common/nRF5_SDK/modules/nrfx/mdk/nrf52840.h\"\n#include \"C:/CGMS/Phoenix2/UnitTesting/Phoenix2_UnitTest/TransmitternRF52/Ceedling/vendor/unity/src/unity.h\"\n\n\n\n\n\n\n\n\n\n\n\n\nvoid setUp(void)\n\n{\n\n\n\n}\n\n\n\nvoid tearDown(void)\n\n{\n\n\n\n}\n\n\n\n\n\n\n\n\n\nvoid test_batt_state_init(void)\n\n{\n\n batt_state_noninit_t * sp_batt_state_noninit_Return;\n\n\n\n    lib_noninit_get_CMockExpectAndReturn(73, 0x74746142, &sp_batt_state_noninit_Return);\n\n (lib_csr_uint_get_CMockExpectAndReturn(74, BATT_DEFAULT_100_PERCENT_MV, 4138));\n\n drv_rtc_monotonic_ticks_get_CMockExpectAndReturn(75, 100);\n\n drv_usbd_power_event_cb_set_CMockIgnore();\n\n\n\n\n\n\n\n batt_state_init();\n\n\n\n\n\n UnityAssertEqualNumber((_U_SINT)((4138)), (_U_SINT)((sp_batt_state_noninit->undocked_100_percent_mv)), (\n\n((void *)0)\n\n), (_U_UINT)83, UNITY_DISPLAY_STYLE_INT);\n\n UnityAssertEqualNumber((_U_SINT)((100)), (_U_SINT)((s_next_log_tickstamp)), (\n\n((void *)0)\n\n), (_U_UINT)84, UNITY_DISPLAY_STYLE_INT);\n\n}\n\n\n\nvoid test_batt_state_simulate_usb_disconnect(void)\n\n{\n\n s_batt_samples.simulate_usb_disconnect = \n\n                                         0\n\n                                              ;\n\n s_batt_samples.is_docked = \n\n                           1\n\n                               ;\n\n\n\n  rtal_mutex_grab_CMockExpect(114, &s_batt_state_mutex);\n\n  hal_gpio_charging_get_CMockExpectAndReturn(115, \n\n 1\n\n );\n\n  drv_rtc_monotonic_ticks_get_CMockExpectAndReturn(116, 100);\n\n   rtal_ms_to_ticks_CMockExpectAndReturn(117, 2000, 2000*1024);\n\n drv_rtc_monotonic_ticks_until_CMockExpectAndReturn(118, 2048100, 1000);\n\n rtal_mutex_release_CMockExpect(119, &s_batt_state_mutex);\n\n app_log_misc_add_CMockIgnore();\n\n\n\n thread_batt_signal_state_change_CMockExpect(122);\n\n thread_mode_post_event_CMockExpect(123, MODE_EVENT_USB_DISCONNECTED);\n\n\n\n\n\n\n\n batt_state_simulate_usb_disconnect(\n\n                                   1\n\n                                       );\n\n\n\n\n\n\n\n}\n\n\n\nvoid test_batt_state_simulate_usb_disconnect_False(void)\n\n{\n\n s_batt_samples.simulate_usb_disconnect = \n\n                                         1\n\n                                             ;\n\n s_batt_samples.is_docked = \n\n                           1\n\n                               ;\n\n\n\n  rtal_mutex_grab_CMockExpect(138, &s_batt_state_mutex);\n\n  hal_gpio_charging_get_CMockExpectAndReturn(139, \n\n 1\n\n );\n\n  drv_rtc_monotonic_ticks_get_CMockExpectAndReturn(140, 100);\n\n   rtal_ms_to_ticks_CMockExpectAndReturn(141, 2000, 2000*1024);\n\n drv_rtc_monotonic_ticks_until_CMockExpectAndReturn(142, 2048100, 0);\n\n (lib_csr_uint_get_CMockExpectAndReturn(143, BATT_UNDOCKED_LOG_INTERVAL_S, 3600));\n\n rtal_secs_to_ticks_CMockExpectAndReturn(144, 3600, 3600*1024);\n\n thread_batt_event_log_CMockExpect(145, BATT_LOG_EVENT_NONE);\n\n\n\n\n\n\n\n rtal_mutex_release_CMockExpect(149, &s_batt_state_mutex);\n\n\n\n  thread_batt_signal_state_change_CMockExpect(151);\n\n  thread_mode_post_event_CMockExpect(152, MODE_EVENT_USB_CONNECTED);\n\n  thread_alarm_post_event_CMockExpect(153, ALARM_DEACTIVATED, ALARM_EC_BATT_TEMP, \n\n ((void *)0)\n\n , \n\n ((void *)0)\n\n );\n\n\n\n\n\n\n\n batt_state_simulate_usb_disconnect(\n\n                                   0\n\n                                        );\n\n\n\n\n\n\n\n}\n\n\n\nvoid test_usbd_power_event_cb_USBD_POWER_EVENT_DISCONNECTED(void)\n\n{\n\n s_batt_samples.simulate_usb_disconnect = \n\n                                         0\n\n                                              ;\n\n s_batt_samples.is_docked = \n\n                           1\n\n                               ;\n\n\n\n  rtal_mutex_grab_CMockExpect(168, &s_batt_state_mutex);\n\n  hal_gpio_charging_get_CMockExpectAndReturn(169, \n\n 1\n\n );\n\n  drv_rtc_monotonic_ticks_get_CMockExpectAndReturn(170, 100);\n\n   rtal_ms_to_ticks_CMockExpectAndReturn(171, 2000, 2000*1024);\n\n drv_rtc_monotonic_ticks_until_CMockExpectAndReturn(172, 2048100, 0);\n\n (lib_csr_uint_get_CMockExpectAndReturn(173, BATT_UNDOCKED_LOG_INTERVAL_S, 3600));\n\n rtal_secs_to_ticks_CMockExpectAndReturn(174, 3600, 3600*1024);\n\n thread_batt_event_log_CMockExpect(175, BATT_LOG_EVENT_NONE);\n\n\n\n\n\n\n\n rtal_mutex_release_CMockExpect(179, &s_batt_state_mutex);\n\n\n\n thread_batt_signal_state_change_CMockExpect(181);\n\n thread_mode_post_event_CMockExpect(182, MODE_EVENT_USB_DISCONNECTED);\n\n (lib_csr_bool_set_CMockExpect(183, PRODUCTION_MODE_ENABLE, \n\n0\n\n));\n\n\n\n\n\n usbd_power_event_cb(USBD_POWER_EVENT_DISCONNECTED);\n\n\n\n\n\n\n\n}\n\n\n\nvoid test_usbd_power_event_cb_USBD_USBD_POWER_EVENT_CONNECTED(void)\n\n{\n\n s_batt_samples.simulate_usb_disconnect = \n\n                                         0\n\n                                              ;\n\n s_batt_samples.is_docked = \n\n                           1\n\n                               ;\n\n\n\n  rtal_mutex_grab_CMockExpect(197, &s_batt_state_mutex);\n\n  hal_gpio_charging_get_CMockExpectAndReturn(198, \n\n 1\n\n );\n\n  drv_rtc_monotonic_ticks_get_CMockExpectAndReturn(199, 100);\n\n   rtal_ms_to_ticks_CMockExpectAndReturn(200, 2000, 2000*1024);\n\n drv_rtc_monotonic_ticks_until_CMockExpectAndReturn(201, 2048100, 0);\n\n (lib_csr_uint_get_CMockExpectAndReturn(202, BATT_UNDOCKED_LOG_INTERVAL_S, 3600));\n\n rtal_secs_to_ticks_CMockExpectAndReturn(203, 3600, 3600*1024);\n\n thread_batt_event_log_CMockExpect(204, BATT_LOG_EVENT_NONE);\n\n\n\n\n\n\n\n rtal_mutex_release_CMockExpect(208, &s_batt_state_mutex);\n\n\n\n thread_batt_signal_state_change_CMockExpect(210);\n\n thread_mode_post_event_CMockExpect(211, MODE_EVENT_USB_CONNECTED);\n\n thread_alarm_post_event_CMockExpect(212, ALARM_DEACTIVATED, ALARM_EC_BATT_TEMP, \n\n((void *)0)\n\n, \n\n((void *)0)\n\n);\n\n\n\n\n\n\n\n usbd_power_event_cb(USBD_POWER_EVENT_CONNECTED);\n\n\n\n\n\n\n\n}\n\n\n\nvoid test_batt_state_get_string(void)\n\n{\n\n\n\nuint8_t p_name;\n\n uint16_t p_offset;\n\n\n\n   strncpy_with_term_CMockIgnore();\n\n\n\n\n\n\n\n batt_state_get_string(BATT_STATE_UNDOCKED_HIGH,&p_name,&p_offset);\n\n\n\n\n\n\n\n}\n\n\n\nvoid test_batt_state_samples_hdr_print(void)\n\n{\n\n\n\nuint8_t p_name;\n\n uint16_t p_offset;\n\n\n\n\n\n\n\n\n\n\n\n batt_state_samples_hdr_print();\n\n\n\n\n\n\n\n}\n\n\n\nvoid test_batt_state_samples_get(void)\n\n{\n\n\n\nbatt_samples_t p_batt_samples;\n\n\n\n\n\n\n\n batt_state_samples_get(&p_batt_samples);\n\n\n\n\n\n\n\n}\n\n\n\nvoid test_batt_state_temp_celsius_get(void)\n\n{\n\n\n\n\n\n     s_batt_samples.acc_temp_celsius = 10.0;\n\n   s_batt_samples.mcu_temp_celsius = 11.0;\n\n\n\n\n\n UnityAssertEqualNumber((_U_SINT)((10.0)), (_U_SINT)((batt_state_temp_celsius_get(BATT_MONITOR_SOURCE_ACC))), (\n\n((void *)0)\n\n), (_U_UINT)275, UNITY_DISPLAY_STYLE_INT);\n\n\n\n\n\n\n\n\n\n\n\n UnityAssertEqualNumber((_U_SINT)((11.0)), (_U_SINT)((batt_state_temp_celsius_get(BATT_MONITOR_SOURCE_MCU))), (\n\n((void *)0)\n\n), (_U_UINT)281, UNITY_DISPLAY_STYLE_INT);\n\n\n\n\n\n  do_p_assert_CMockExpect(284, \n\n 0\n\n , \"../Application/batt_state.c\", 767);\n\n\n\n\n\n batt_state_temp_celsius_get(BATT_MONITOR_MAX);\n\n\n\n\n\n\n\n\n\n}\n\n\n\n\n\n\n\nvoid test_batt_state_avg_mv_get(void)\n\n{\n\n\n\n s_batt_samples.batt_avg_mv = 3;\n\n\n\n\n\n\n\n\n\n UnityAssertEqualNumber((_U_SINT)((3)), (_U_SINT)((batt_state_avg_mv_get())), (\n\n((void *)0)\n\n), (_U_UINT)304, UNITY_DISPLAY_STYLE_INT);\n\n\n\n\n\n\n\n}\n\nvoid test_batt_state_enter(void)\n\n{\n\n\n\n\n\n\n\n batt_state_enter(BATT_STATE_UNDOCKED_HIGH);\n\n\n\n\n\n batt_state_enter(BATT_STATE_UNDOCKED_UNKNOWN);\n\n thread_alarm_post_event_CMockIgnore();\n\n\n\n do_p_assert_CMockExpect(377, \n\n0\n\n, \"../Application/batt_state.c\", 514);\n\n do_p_assert_CMockExpect(378, \n\n0\n\n, \"../Application/batt_state.c\", 291);\n\n (lib_csr_uint_get_CMockIgnoreAndReturn(379, 0));\n\n app_alarm_core_is_active_or_deferred_CMockIgnoreAndReturn(380, \n\n1\n\n);\n\n\n\n\n\n batt_state_enter(BATT_STATE_UNDOCKED_LOW);\n\n\n\n thread_led_play_CMockExpect(385, (3), (5), g_led_seq_docked_batt_unknown);\n\n\n\n\n\n batt_state_enter(BATT_STATE_DOCKED_UNKNOWN);\n\n\n\n\n\n thread_led_play_CMockExpect(391, (3), (5), g_led_seq_undervoltage_repeat);\n\n\n\n\n\n\n\n  batt_state_enter(BATT_STATE_DOCKED_LOW);\n\n\n\n     thread_led_play_CMockExpect(397, (3), (5), g_led_seq_docked_batt_high);\n\n\n\n\n\n\n\n  batt_state_enter(BATT_STATE_DOCKED_HIGH);\n\n\n\n thread_led_play_CMockExpect(403, (3), (5), g_led_seq_docked_batt_full);\n\n\n\n\n\n\n\n  batt_state_enter(BATT_STATE_DOCKED_FULL);\n\n\n\n\n\n  batt_state_enter(BATT_STATE_MAX);\n\n\n\n\n\n}\n\n\n\nvoid test_batt_state_exit(void)\n\n{\n\n\n\n\n\n s_batt_state = BATT_STATE_UNDOCKED_HIGH;\n\n\n\n batt_state_exit(BATT_STATE_UNDOCKED_HIGH);\n\n\n\n\n\n s_batt_state = BATT_STATE_DOCKED_UNKNOWN;\n\n thread_led_stop_CMockExpect(425, (3));\n\n\n\n\n\n batt_state_exit(BATT_STATE_DOCKED_UNKNOWN);\n\n\n\n s_batt_state = BATT_STATE_MAX;\n\n\n\n batt_state_exit(BATT_STATE_MAX);\n\n}\n\n\n\n\n\nvoid test_new_batt_state_to_log_event(void)\n\n{\n\n\n\n\n\n s_batt_state = BATT_STATE_DOCKED_UNKNOWN;\n\n\n\n UnityAssertEqualNumber((_U_SINT)((BATT_LOG_EVENT_DOCKED)), (_U_SINT)((new_batt_state_to_log_event())), (\n\n((void *)0)\n\n), (_U_UINT)442, UNITY_DISPLAY_STYLE_INT);\n\n\n\n\n\n s_batt_state = BATT_STATE_UNDOCKED_UNKNOWN;\n\n\n\n UnityAssertEqualNumber((_U_SINT)((BATT_LOG_EVENT_UNDOCKED)), (_U_SINT)((new_batt_state_to_log_event())), (\n\n((void *)0)\n\n), (_U_UINT)447, UNITY_DISPLAY_STYLE_INT);\n\n\n\n}\n\n\n\nvoid test_batt_state_docked_state_get(void)\n\n{\n\n\n\n\n\n s_batt_samples.is_charging =\n\n                            0\n\n                                 ;\n\n\n\n UnityAssertEqualNumber((_U_SINT)((BATT_STATE_DOCKED_FULL)), (_U_SINT)((batt_state_docked_state_get())), (\n\n((void *)0)\n\n), (_U_UINT)457, UNITY_DISPLAY_STYLE_INT);\n\n\n\n\n\n  s_batt_samples.is_charging =\n\n                             1\n\n                                 ;\n\n  s_batt_samples.batt_mv = 4030;\n\n\n\n UnityAssertEqualNumber((_U_SINT)((BATT_STATE_DOCKED_HIGH)), (_U_SINT)((batt_state_docked_state_get())), (\n\n((void *)0)\n\n), (_U_UINT)463, UNITY_DISPLAY_STYLE_INT);\n\n\n\n s_batt_samples.is_charging =\n\n                            1\n\n                                ;\n\n  s_batt_samples.batt_mv = 4026;\n\n\n\n UnityAssertEqualNumber((_U_SINT)((BATT_STATE_DOCKED_LOW)), (_U_SINT)((batt_state_docked_state_get())), (\n\n((void *)0)\n\n), (_U_UINT)468, UNITY_DISPLAY_STYLE_INT);\n\n\n\n}\n\n\n\nvoid test_batt_state_docking_change_handle(void)\n\n{\n\n s_batt_samples.is_charging = \n\n                             1\n\n                                 ;\n\n    s_batt_samples.batt_mv = 100;\n\n    s_batt_samples.batt_avg_mv = 100;\n\n    s_batt_samples.undocked_100_percent_mv = 4328;\n\n    s_batt_samples.batt_level_percent = 10;\n\n s_batt_samples.is_docked = \n\n                           1\n\n                               ;\n\n\n\n\n\n\n\n UnityAssertEqualNumber((_U_SINT)((BATT_STATE_DOCKED_UNKNOWN)), (_U_SINT)((batt_state_docking_change_handle())), (\n\n((void *)0)\n\n), (_U_UINT)483, UNITY_DISPLAY_STYLE_INT);\n\n\n\n\n\n UnityAssertEqualNumber((_U_SINT)((s_batt_samples.is_charging)), (_U_SINT)((\n\n0\n\n)), (\n\n((void *)0)\n\n), (_U_UINT)486, UNITY_DISPLAY_STYLE_INT);\n\n    UnityAssertEqualNumber((_U_SINT)((s_batt_samples.batt_mv)), (_U_SINT)((0)), (\n\n   ((void *)0)\n\n   ), (_U_UINT)487, UNITY_DISPLAY_STYLE_INT);\n\n    UnityAssertEqualNumber((_U_SINT)((s_batt_samples.batt_avg_mv)), (_U_SINT)((0)), (\n\n   ((void *)0)\n\n   ), (_U_UINT)488, UNITY_DISPLAY_STYLE_INT);\n\n    UnityAssertEqualNumber((_U_SINT)((s_batt_samples.undocked_100_percent_mv)), (_U_SINT)((0)), (\n\n   ((void *)0)\n\n   ), (_U_UINT)489, UNITY_DISPLAY_STYLE_INT);\n\n    UnityAssertEqualNumber((_U_SINT)((s_batt_samples.batt_level_percent)), (_U_SINT)((0)), (\n\n   ((void *)0)\n\n   ), (_U_UINT)490, UNITY_DISPLAY_STYLE_INT);\n\n if (!(s_batt_samples.charging_completed)) {} else {UnityFail( (\" Expected FALSE Was TRUE\"), (_U_UINT)(_U_UINT)491);;};\n\n  UnityAssertEqualNumber((_U_SINT)((s_batt_level_percent_ratchet)), (_U_SINT)((0)), (\n\n ((void *)0)\n\n ), (_U_UINT)492, UNITY_DISPLAY_STYLE_INT);\n\n\n\n}\n\n\n\nvoid test_batt_state_mv_measure(void)\n\n{\n\n\n\n\n\n s_batt_samples.is_docked =\n\n                          1\n\n                              ;\n\n s_avg_sample_count =0;\n\n s_avg_adc_mv = 1;\n\n\n\n   hal_adc_battery_mv_read_CMockExpectAndReturn(504, 3);\n\n   (lib_csr_bool_get_CMockExpectAndReturn(505, ENABLE_BATTERY_LEVEL_SIMULATE, \n\n  0\n\n  ));\n\n   (lib_csr_uint_get_CMockExpectAndReturn(506, BATT_MV_RMA_WINDOW_S, 150));\n\n   (lib_csr_uint_get_CMockExpectAndReturn(507, BATT_DOCKED_SAMPLE_INTERVAL_S, 1));\n\n\n\n\n\n batt_state_mv_measure();\n\n\n\n UnityAssertEqualNumber((_U_SINT)((s_avg_sample_count)), (_U_SINT)((1)), (\n\n((void *)0)\n\n), (_U_UINT)512, UNITY_DISPLAY_STYLE_INT);\n\n UnityAssertEqualNumber((_U_SINT)((s_avg_adc_mv)), (_U_SINT)((3)), (\n\n((void *)0)\n\n), (_U_UINT)513, UNITY_DISPLAY_STYLE_INT);\n\n UnityAssertEqualNumber((_U_SINT)((s_batt_samples.batt_avg_mv)), (_U_SINT)((3.0)), (\n\n((void *)0)\n\n), (_U_UINT)514, UNITY_DISPLAY_STYLE_INT);\n\n\n\n}\n\n\n\nvoid test_batt_state_mv_measure_ENABLE_BATTERY_LEVEL_SIMULATE_true(void)\n\n{\n\n\n\n\n\n s_batt_samples.is_docked =\n\n                          0\n\n                               ;\n\n s_avg_sample_count =0;\n\n s_avg_adc_mv = 1;\n\n\n\n   hal_adc_battery_mv_read_CMockExpectAndReturn(526, 4280);\n\n   (lib_csr_bool_get_CMockExpectAndReturn(527, ENABLE_BATTERY_LEVEL_SIMULATE, \n\n  1\n\n  ));\n\n   (lib_csr_uint_get_CMockExpectAndReturn(528, BATT_SIMULATION_MV, 3900));\n\n   (lib_csr_uint_get_CMockExpectAndReturn(529, BATT_MV_RMA_WINDOW_S, 150));\n\n   (lib_csr_uint_get_CMockExpectAndReturn(530, BATT_UNDOCKED_SAMPLE_INTERVAL_S, 15));\n\n\n\n\n\n batt_state_mv_measure();\n\n\n\n UnityAssertEqualNumber((_U_SINT)((s_avg_sample_count)), (_U_SINT)((1)), (\n\n((void *)0)\n\n), (_U_UINT)535, UNITY_DISPLAY_STYLE_INT);\n\n UnityAssertEqualNumber((_U_SINT)((s_avg_adc_mv)), (_U_SINT)((3900)), (\n\n((void *)0)\n\n), (_U_UINT)536, UNITY_DISPLAY_STYLE_INT);\n\n UnityAssertEqualNumber((_U_SINT)((s_batt_samples.batt_avg_mv)), (_U_SINT)((3900.0)), (\n\n((void *)0)\n\n), (_U_UINT)537, UNITY_DISPLAY_STYLE_INT);\n\n\n\n}\n\n\n\nvoid test_batt_batt_state_temp_measure(void)\n\n{\n\n\n\n\n\n\n\n\n\n   drv_accel_temp_celsius_get_CMockIgnoreAndReturn(547, 3);\n\n   hal_adc_mcu_temp_celsius_read_CMockExpectAndReturn(548, 2.0);\n\n\n\n\n\n batt_state_temp_measure();\n\n\n\n UnityAssertEqualNumber((_U_SINT)((s_batt_samples.mcu_temp_celsius)), (_U_SINT)((2.0)), (\n\n((void *)0)\n\n), (_U_UINT)553, UNITY_DISPLAY_STYLE_INT);\n\n\n\n}\n\n\n\n\n\nvoid test_batt_state_percent_calc(void)\n\n{\n\n\n\n\n\n s_batt_samples.is_docked =\n\n                          1\n\n                              ;\n\n s_batt_samples.is_charging = \n\n                             0\n\n                                  ;\n\n\n\n   (lib_csr_uint_get_CMockExpectAndReturn(565, BATT_DEFAULT_0_PERCENT_MV, 3722));\n\n   (lib_csr_uint_get_CMockExpectAndReturn(566, BATT_DEFAULT_100_PERCENT_MV, 4138));\n\n\n\n\n\n\n\n batt_state_percent_calc(\n\n                        0\n\n                             );\n\n\n\n if ((s_batt_samples.charging_completed)) {} else {UnityFail( (\" Expected TRUE Was FALSE\"), (_U_UINT)(_U_UINT)572);;};\n\n\n\n}\n\n\n\nvoid test_batt_state_percent_calc_is_docked_false(void)\n\n{\n\n\n\n\n\n s_batt_samples.is_docked =\n\n                          0\n\n                               ;\n\n s_batt_samples.is_charging = \n\n                             0\n\n                                  ;\n\n s_batt_samples.batt_mv = 4100;\n\n\n\n  (lib_csr_uint_get_CMockExpectAndReturn(584, BATT_DEFAULT_0_PERCENT_MV, 3722));\n\n   (lib_csr_uint_get_CMockExpectAndReturn(585, BATT_DEFAULT_100_PERCENT_MV, 4138));\n\n\n\n\n\n\n\n batt_state_percent_calc(\n\n                        1\n\n                            );\n\n\n\n if ((s_batt_samples.charging_completed)) {} else {UnityFail( (\" Expected TRUE Was FALSE\"), (_U_UINT)(_U_UINT)591);;};\n\n\n\n}\n":String
C:/CGMS/Phoenix2/UnitTesting/Phoenix2_UnitTest/TransmitternRF52/Ceedling/vendor/unity/auto/generate_test_runner.rb:73:in `find_tests'
C:/CGMS/Phoenix2/UnitTesting/Phoenix2_UnitTest/TransmitternRF52/Ceedling/lib/ceedling/generator_test_runner.rb:19:in `find_test_cases'
C:/CGMS/Phoenix2/UnitTesting/Phoenix2_UnitTest/TransmitternRF52/Ceedling/lib/ceedling/generator.rb:65:in `generate_test_runner'
C:/CGMS/Phoenix2/UnitTesting/Phoenix2_UnitTest/TransmitternRF52/Ceedling/lib/ceedling/rules_tests.rake:8:in `block in <top (required)>'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/rake-13.0.3/lib/rake/task.rb:281:in `block in execute'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/rake-13.0.3/lib/rake/task.rb:281:in `each'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/rake-13.0.3/lib/rake/task.rb:281:in `execute'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/rake-13.0.3/lib/rake/task.rb:219:in `block in invoke_with_call_chain'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/rake-13.0.3/lib/rake/task.rb:199:in `synchronize'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/rake-13.0.3/lib/rake/task.rb:199:in `invoke_with_call_chain'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/rake-13.0.3/lib/rake/task.rb:188:in `invoke'
C:/CGMS/Phoenix2/UnitTesting/Phoenix2_UnitTest/TransmitternRF52/Ceedling/lib/ceedling/task_invoker.rb:67:in `invoke_test_runner'
C:/CGMS/Phoenix2/UnitTesting/Phoenix2_UnitTest/TransmitternRF52/Ceedling/lib/ceedling/test_invoker.rb:107:in `block in setup_and_invoke'
C:/CGMS/Phoenix2/UnitTesting/Phoenix2_UnitTest/TransmitternRF52/Ceedling/lib/ceedling/test_invoker.rb:51:in `each'
C:/CGMS/Phoenix2/UnitTesting/Phoenix2_UnitTest/TransmitternRF52/Ceedling/lib/ceedling/test_invoker.rb:51:in `setup_and_invoke'
C:/CGMS/Phoenix2/UnitTesting/Phoenix2_UnitTest/TransmitternRF52/Ceedling/lib/ceedling/rules_tests.rake:70:in `block (2 levels) in <top (required)>'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/rake-13.0.3/lib/rake/task.rb:281:in `block in execute'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/rake-13.0.3/lib/rake/task.rb:281:in `each'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/rake-13.0.3/lib/rake/task.rb:281:in `execute'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/rake-13.0.3/lib/rake/task.rb:219:in `block in invoke_with_call_chain'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/rake-13.0.3/lib/rake/task.rb:199:in `synchronize'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/rake-13.0.3/lib/rake/task.rb:199:in `invoke_with_call_chain'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/rake-13.0.3/lib/rake/task.rb:188:in `invoke'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/rake-13.0.3/lib/rake/application.rb:160:in `invoke_task'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/rake-13.0.3/lib/rake/application.rb:116:in `block (2 levels) in top_level'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/rake-13.0.3/lib/rake/application.rb:116:in `each'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/rake-13.0.3/lib/rake/application.rb:116:in `block in top_level'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/rake-13.0.3/lib/rake/application.rb:125:in `run_with_threads'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/rake-13.0.3/lib/rake/application.rb:110:in `top_level'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/rake-13.0.3/lib/rake/application.rb:83:in `block in run'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/rake-13.0.3/lib/rake/application.rb:186:in `standard_exception_handling'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/rake-13.0.3/lib/rake/application.rb:80:in `run'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/rake-13.0.3/exe/rake:27:in `<top (required)>'
C:/Ruby30-x64/bin/rake:33:in `load'
C:/Ruby30-x64/bin/rake:33:in `<main>'
Tasks: TOP => build/test/runners/test_batt_state_runner.c

Thanks
Suresh
